api/data_explorer/util/elasticsearch_util.py

```diff
@@ -215,13 +251,39 @@ def get_elasticsearch_facet(es, elasticsearch_field_name, field_type):
             field=elasticsearch_field_name,
             interval=_get_bucket_interval(field_range))
 
-    # Handle sample facets in a special way since they are nested objects.
-    if elasticsearch_field_name.startswith('samples.'):
-        es_facet = NestedFacet('samples', es_facet)
-
+    nested_facet = _get_nested_facet(elasticsearch_field_name, es_facet,
+                                     nested_paths)
+    if nested_facet:
+        es_facet = nested_facet
     return es_facet
 
 
+def _get_nested_facet(elasticsearch_field_name, es_facet, nested_paths):
```

Comment:
I think _maybe_get_nested_facet is more clear

Reply:
Done

Reply:
Not done, did you forget to push?

Reply:
Oh, didn't see that "maybe" was part of the method name.